services/port_manager/src/main/java/com/futurewei/alcor/portmanager/processor/FixedIpsProcessor.java

```diff
@@ -107,6 +115,143 @@ private void releaseFixedIpAddress(PortContext context, List<IpAddrRequest> ipAd
         }
     }
 
+    /**
+     * Fill in information to PortEntity
+     * @param context
+     * @param ipAddrUpdateRequest
+     * @throws Exception
+     */
+    private void updateFixedIpAddress(PortContext context,IpAddrUpdateRequest ipAddrUpdateRequest) throws Exception{
+        UpdatePortIpAddressRequest updatePortIpAddressRequest = new UpdatePortIpAddressRequest(context, ipAddrUpdateRequest);
+        context.getRequestManager().sendRequest(updatePortIpAddressRequest);
+
+        List<IpAddrUpdateRequest> result = updatePortIpAddressRequest.getResult();
+        context.setResult(result);
+
+        List<String> filterIpsHasSubnet = context.getHasSubnetFixedIps().stream()
+                .map(PortEntity.FixedIp::getIpAddress)
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+
+        List<String> filterIpsHasIp = context.getHasIpFixedIps().values().stream()
+                .map(PortEntity.FixedIp::getIpAddress)
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+
+        for (IpAddrUpdateRequest request : result) {
+            Iterator<IpAddrRequest> iterator = request.getOldIpAddrRequests().iterator();
+            while (iterator.hasNext()){
+                IpAddrRequest ipAddr = iterator.next();
+
+                if(filterIpsHasIp.contains(ipAddr.getIp())){
+                    for (PortEntity.FixedIp fixedIp : context.getOldPortEntity().getFixedIps()) {
+                        if(ipAddr.getIp().equals(fixedIp.getIpAddress())){
+                            fixedIp.setIpV4RangeId(ipAddr.getRangeId());
+                            fixedIp.setSubnetId(ipAddr.getSubnetId());
+                        }
+                    }
+                    for (Map.Entry<String, PortEntity.FixedIp> fixedIpEntry : context.getHasIpFixedIps().entrySet()) {
+                        if(ipAddr.getIp().equals(fixedIpEntry.getKey())){
+                            fixedIpEntry.getValue().setIpV4RangeId(ipAddr.getRangeId());
+                            fixedIpEntry.getValue().setSubnetId(ipAddr.getSubnetId());
+                        }
+                    }
+                    iterator.remove();
+                    continue;
+                }
+
+                if(filterIpsHasSubnet.contains(ipAddr.getIp())){
+                    iterator.remove();
+                    continue;
+                }
+
+                for (PortEntity.FixedIp fixedIp : context.getHasSubnetFixedIps()) {
+                    if(ipAddr.getSubnetId().equals(fixedIp.getSubnetId())){
+                        fixedIp.setIpAddress(ipAddr.getIp());
```

Comment:
If HasSubnetFixedIps.size () > 1 and all have the same subnetId, will all FixedIp have the same ipAddr?

Reply:
I will delete it  because it have no use in the later code